store/src/file_sink.rs

```diff
@@ -295,3 +295,64 @@ impl FileSink {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use async_compression::tokio::bufread::GzipDecoder;
+    use futures::stream::StreamExt;
+    use tokio::{fs::DirEntry, io::BufReader};
+    use tokio_util::codec::{length_delimited::LengthDelimitedCodec, FramedRead};
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn writes_a_framed_gzip_encoded_file() {
+        let tmp_dir = tempdir::TempDir::new("entropy_tests").unwrap();
+        let (shutdown_trigger, shutdown_listener) = triggered::trigger();
+        let (sender, receiver) = message_channel(10);
+
+        let mut file_sink = FileSinkBuilder::new(FileType::Entropy, &tmp_dir.path(), receiver)
+            .roll_time(chrono::Duration::milliseconds(100))
+            .create()
+            .await
+            .unwrap();
```

Comment:
```suggestion
            .await?;
```

Reply:
You can't do the question mark bit here since a test does not return a result. The convention is to use `.expect("could not create sink")` to make the failure clear

Reply:
we worked out that a test is similar to `main` in that it doesn't return a result by default; i sort of understand how that works in the sense of a test being a standalone execution of the code in question but on the other hand it seems so natural to me that a test should return a `Result`

Reply:
Addressed in dc8b27e